src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      resolved = true;
+      resolve('First promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+
+  setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  const handleClick = (e) => {
+    e.preventDefault();
```

Comment:
The `e.preventDefault();` call here might prevent default actions for all clicks, which may not be intended. Consider whether this is necessary for all clicks or just specific ones.